### PR TITLE
Elasticsearch/Kibana/Fluentd Configuration

### PR DESCRIPTION
**Pull request by johnc1231**
This PR sets up Elasticsearch, Kibana, and Fluentd on the Kubernetes cluster, allowing us to go to internal.hail.is/monitoring/kibana and look at logs from the cluster.

### CODE REVIEW

File: monitoring/monitoring.yaml [hunk 1]
```diff
@@ -416,3 +416,259 @@ spec:
...
+              mountPath: /var/lib/docker/containers
+              readOnly: true
+      terminationGracePeriodSeconds: 30
+      volumes:
+        - name: varlog
+          hostPath:
+            path: /var/log
+        - name: varlibdockercontainers
+          hostPath:
+            path: /var/lib/docker/containers
```
Comment (danking) [inline monitoring/monitoring.yaml]: remove extra trailing space

File: monitoring/monitoring.yaml [hunk 2]
```diff
@@ -416,3 +416,259 @@ spec:
...
+      app: fluentd
+  template:
+    metadata:
+      labels:
+        app: fluentd
+    spec:
+      serviceAccount: fluentd
+      serviceAccountName: fluentd
+      tolerations:
+        - key: node-rolde.kubernetes.io/master # This toleration makes Fluentd pod also run on master node.
```
Comment (danking) [inline monitoring/monitoring.yaml]: is it really `node-rolde`?

File: monitoring/monitoring.yaml [hunk 3]
```diff
@@ -416,3 +416,259 @@ spec:
...
+        - name: fix-permissions
+          image: busybox
+          command: ["sh", "-c", "chown -R 1000:1000 /usr/share/elasticsearch/data"]
+          securityContext:
+            privileged: true
+          volumeMounts:
+            - name: data
+              mountPath: /usr/share/elasticsearch/data
+        - name: increase-vm-max-map
+          image: busybox
```
Comment (danking) [inline monitoring/monitoring.yaml]: prefer pinning image versions

File: monitoring/monitoring.yaml [hunk 4]
```diff
@@ -416,3 +416,259 @@ spec:
...
+spec:
+  selector:
+    matchLabels:
+      app: fluentd
+  template:
+    metadata:
+      labels:
+        app: fluentd
+    spec:
+      serviceAccount: fluentd
```
Comment (danking) [inline monitoring/monitoring.yaml]: this is deprecated and replaced with the below one, let's remove it

File: monitoring/monitoring.yaml (hunk 2 above)
Comment (johnc1231) [reply monitoring/monitoring.yaml]: Nope, good eye

File: monitoring/monitoring.yaml [hunk 5]
```diff
@@ -416,3 +416,255 @@ spec:
...
+              mountPath: /usr/share/elasticsearch/data
+          env:
+            - name: cluster.name # Should this name be hardcoded?
+              value: gke_hail-vdc_us-central1-a_vdc
+            - name: node.name
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: discovery.zen.ping.unicast.hosts
+              value: "es-cluster-0.elasticsearch,es-cluster-1.elasticsearch,es-cluster-2.elasticsearch" #TODO This seems dumb, why must I list out all numbers?
```
Comment (danking) [inline monitoring/monitoring.yaml]: 1. no comments
2. Normally our build.yaml's are run through a templating engine (when used in build.yaml) http://jinja.pocoo.org/docs/2.10/templates/ Jinja has loops that would let you express this as:
```
{{ ["%s0%s", "%s1%s", "%s2%s"] | format("es-cluster-", ".elasticsearch") | join(",") }}
```
But honestly, the replication you have seems easier to read.

File: monitoring/monitoring.yaml [hunk 6]
```diff
@@ -416,3 +416,255 @@ spec:
...
+            - name: discovery.zen.minimum_master_nodes
+              value: "2" #Should be N/2 + 1 at minimum, where N is the number of replicas.
```
Comment (danking) [inline monitoring/monitoring.yaml]: again, with Jinja you could do 

```
{% set n_replicas = 3 %}
...
replicas: {{ n_replicas }}
...
          value: {{ n_replicas // 2 + 1 }}
```

Though I like the parameter to be failure_tolerance and set replicas to failure_tolerance * 2 + 1

File: monitoring/monitoring.yaml [hunk 7]
```diff
@@ -416,3 +416,255 @@ spec:
...
+            - name: cluster.initial_master_nodes
+              value: "es-cluster-0,es-cluster-1,es-cluster-2"
+            - name: ES_JAVA_OPTS
+              value: "-Xms512m -Xmx512m" #Just grabbed numbers from tutorial
```
Comment (danking) [inline monitoring/monitoring.yaml]: no comments, unless its a link to an SO post or something that explains it.

File: monitoring/monitoring.yaml [hunk 8]
```diff
@@ -416,3 +416,255 @@ spec:
...
+    matchLabels:
+      app: fluentd
+  template:
+    metadata:
+      labels:
+        app: fluentd
+    spec:
+      serviceAccountName: fluentd
+      tolerations:
+        - key: node-role.kubernetes.io/master # This toleration makes Fluentd pod also run on master node.
```
Comment (danking) [inline monitoring/monitoring.yaml]: no comment, that's pretty clear from the toleration name if you understand tolerations. Understanding of tolerations is an OK prerequisite to assume of anyone modifying this file.

**Pull request merged**